cmd/kpod/create.go

```diff
@@ -0,0 +1,192 @@
+package main
+
+import (
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+	pb "k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime"
+)
+
+// TODO: Missing flags from docker-create - particularly --security-opt or equiv and resource limit flags
+// TODO stop using Kubernetes API structs here - replace with our own versions (remove pb import entirely)
+// TODO Add missing flags from docker-create
+// TODO parse flags into a createConfig, and parse createConfig into an OCI runtime spec
+
+// TODO These temporary values should be replaced with sane defaults
+const (
+	defaultHostname     = "kpod-launch"
+	defaultCgroupParent = "/kpod-launch"
+)
+
+type createConfig struct {
+	image            string
+	command          string
+	args             []string
+	pod              string
+	privileged       bool
+	rm               bool
+	hostNet          bool
+	hostPID          bool
+	hostIPC          bool
+	name             string
+	labels           map[string]string
+	workDir          string
+	env              map[string]string
+	detach           bool
+	stdin            bool
+	tty              bool
+	devices          []*pb.Device
+	mounts           []*pb.Mount
+	capAdd           []string
+	capDrop          []string
+	dnsServers       []string
+	dnsSearch        []string
+	dnsOpt           []string
+	ports            []*pb.PortMapping
+	hostname         string
+	cgroupParent     string
+	sysctl           string
+	user             int64
+	additionalGroups []int64
+	readOnlyRootfs   bool
+}
+
+var createFlags = []cli.Flag{
+	cli.StringFlag{
+		Name:  "pod",
```

Comment:
Drop

Reply:
Don't we want to keep this? We need to add containers to pods when they're created.

Reply:
Ok, your right, I was thinking this was creating a pod.  I guess I should have read the comment.

Reply:
Which brings up an interesting thought, should we always create a pod if a user does not specify one?

Reply:
libpod won't require containers to be in a pod, so I don't think we need to/want to. Pods are only required if you want to share namespaces with other containers.

The only potential issue with this is I don't presently allow containers to be added to a pod after they're created, so if we don't make them a new pod we won't be able to join other containers to their namespaces after the fact.

Reply:
Well we need to support all of the --pid container:ID calls, which would allow you to do this manually.